crates/transpiler/src/passes/commutative_optimization.rs

```diff
@@ -393,6 +384,17 @@ fn try_merge(
     Ok((false, None, 0.))
 }
 
+/// Returns whether qubits/clbits for one instruction are fully disjoint from qubit/clbits of
+/// another instruction.
+fn disjoint_instructions(
```

Comment:
We could probably inline this since it's a tiny function that's called repeatedly 
```suggestion
/// another instruction.
#[inline]
fn disjoint_instructions(
```